app/model/application.js

```diff
@@ -438,6 +284,25 @@ ApplicationSchema.statics.updateBuild = function(data) {
   });
 }
 
+ApplicationSchema.pre('save', next => {
+  console.log(this);
+
+//  if (this.isNew) {
+//    return update(this)
+//    .then(messages => {
+//      this.issue.problems = messages;
+//    }, err => {
+//      app.log.error(`A testing error occured with ${application.github.fullName}`);
+//      app.log.error(err);
+//    })
+//    .then(() => {
+//      return next();
+//    })
+//  }
+
+  return next();
```

Comment:
Can someone help me on this. For some reason `this` is always undefined, and in every example / documentation I could find, it should reference the object about to be saved.